contracts/psm/PSM.sol

```diff
@@ -0,0 +1,293 @@
+        sellFee = _sellFee;
+        buyFee = _buyFee;
+        feeReceiver = _feeReceiver;
+        lisUSD = _lisUSD;
+
+        dailyLimit = _dailyLimit;
+        minSell = _minSell;
+        minBuy = _minBuy;
+
+        emit SetBuyFee(_buyFee);
+        emit SetSellFee(_sellFee);
+        emit SetFeeReceiver(_feeReceiver);
+        emit SetDailyLimit(_dailyLimit);
+        emit SetMinSell(_minSell);
+        emit SetMinBuy(_minBuy);
+    }
+
+    /**
+     * @dev sell token to get lisUSD
+     * @param amount token amount
+     */
+    function sell(uint256 amount) external nonReentrant whenNotPaused {
+        // calculate fee and real amount
+        uint256 fee = Math.mulDiv(amount, sellFee, FEE_PRECISION);
+        uint256 realAmount = amount - fee;
+
+        // check sell limit
+        require(amount <= IERC20(lisUSD).balanceOf(address(this)), "exceed sell limit");
+
+        // transfer token from user
+        IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
+
+        // transfer lisUSD to user
+        IERC20(lisUSD).safeTransfer(msg.sender, realAmount);
+
+        // deposit token to vault manager
+        IERC20(token).safeIncreaseAllowance(vaultManager, amount);
+        IVaultManager(vaultManager).deposit(amount);
+
+        // transfer fee to fee receiver
+        if (fee > 0) {
+            IERC20(lisUSD).transfer(feeReceiver, fee);
+        }
+        emit SellToken(msg.sender, realAmount, fee);
+    }
+
+    /**
+     * @dev buy token with lisUSD
+     * @param amount lisUSD amount
+     */
+    function buy(uint256 amount) external nonReentrant whenNotPaused {
+
+        // calculate fee and real amount
+        uint256 fee = Math.mulDiv(amount, buyFee, FEE_PRECISION);
+        uint256 realAmount = amount - fee;
+
+        // check buy limit
+        checkAndUpdateBuyUsed(realAmount);
+
+        // transfer lisUSD from user and withdraw token from vault manager
+        if (realAmount > 0) {
+            IERC20(lisUSD).safeTransferFrom(msg.sender, address(this), realAmount);
+            IVaultManager(vaultManager).withdraw(msg.sender, realAmount);
+        }
+
+        // transfer fee to fee receiver
+        if (fee > 0) {
+            IERC20(lisUSD).safeTransferFrom(msg.sender, feeReceiver, fee);
+        }
+        emit BuyToken(msg.sender, realAmount, fee);
+    }
+
+    // check buy limit
+    function checkBuyLimit(uint256 amount) public view returns (bool) {
+        // check min buy amount
+        if (amount < minBuy) {
+            return false;
+        }
+        // check daily buy limit
+        if (getDay() == lastBuyDay && dayBuyUsed + amount > dailyLimit) {
+            return false;
+        }
+        if (getDay() != lastBuyDay && amount > dailyLimit) {
+            return false;
+        }
+        return true;
+    }
+
+    // check and update buy used
+    function checkAndUpdateBuyUsed(uint256 amount) private {
+        require(checkBuyLimit(amount), "exceed buy limit");
```

Comment:
Need to revise the description message?  amount smaller than minBuy or exceed buy limit"